### PR TITLE
Insert some useful metadatas into the database

### PR DESCRIPTION
**Pull request by paulineMig**
-When writing a dicom file study date, study time, series times and series dates are needed if they are not provided by the data the writer uses the current time but the fractional part of a second could change between slices which can be problematic for some software like lead-DBS used by RebrAIn. Indeed for lead-DBS slices are not stacked if these values change.

-the kvp metadata information was not kept inside the database so the dicom writer was adding 1 by default to all ct dicom 

-Morever we now read patient age from the database so this information is accessible from the medinria metadata list interface

### CODE REVIEW

File: src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp [hunk 1]
```diff
@@ -107,16 +107,30 @@ bool medLocalDbController::createStudyTable()
...
+        " id        INTEGER      PRIMARY KEY,"
+        " patient   INTEGER," // FOREIGN KEY
+        " name         TEXT,"
+        " uid          TEXT,"
+        " thumbnail    TEXT,"
+        " studyId      TEXT,"
+        " time         TEXT,"
+        " date         TEXT"
+        ");") &&
            execQuery(query, __FILE__, __LINE__);
```
Comment (mathildemerle) [inline src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp:121]: it's also done line 125, is it normal?

File: src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp [hunk 2]
```diff
@@ -107,16 +107,30 @@ bool medLocalDbController::createStudyTable()
...
+    query.prepare("PRAGMA table_info(study)");
+    if (!execQuery(query, __FILE__, __LINE__))
+    {
+        qDebug() << DTK_COLOR_FG_RED << query.lastError() << DTK_NO_COLOR;
+    }
+
+    this->addTextColumnToStudyTableIfNeeded(query, "time");
+    this->addTextColumnToStudyTableIfNeeded(query, "date");
+
+    return true;
```
Comment (mathildemerle) [inline src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp:133]: the function never return an error whereas it was possible before. Is it the behavior you were searching to create?

File: src/plugins/legacy/itkDataImage/writers/itkDicomDataImageWriter.cpp [hunk 3]
```diff
@@ -107,72 +107,77 @@ QString itkDicomDataImageWriter::sopClassUID(QString modality)
 
 void itkDicomDataImageWriter::fillDictionaryFromMetaDataKey(itk::MetaDataDictionary &dictionary, bool &studyUIDExistance)
 {
+    bool studyTimeExistance = false;
```
Comment (mathildemerle) [inline src/plugins/legacy/itkDataImage/writers/itkDicomDataImageWriter.cpp:110]: *existing

**Review by mathildemerle — CHANGES_REQUESTED** [review]
I am testing this PR, i've got some errors trying to open in a view an old DICOM i just imported from Browser with this PR. I'm going to continue my tests, but this PR needs to be compatible with old DICOMs

File: src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp (hunk 1 above)
Comment (paulineMig) [reply src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp:121]: I want the same comportement as createSeriesTable() so I copied the code from that function

File: src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp (hunk 2 above)
Comment (paulineMig) [reply src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp:133]: Same as previous comment

**Comment by paulineMig** [comment]
> I am testing this PR, i've got some errors trying to open in a view an old DICOM i just imported from Browser with this PR. I'm going to continue my tests, but this PR needs to be compatible with old DICOMs

can you send me the errors beacuse it seems to be working for me

File: src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp (hunk 1 above)
Comment (paulineMig) [reply src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp:121]: I want the same comportement as createSeriesTable() so I copied the code from that function

File: src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp (hunk 2 above)
Comment (paulineMig) [reply src/layers/legacy/medCoreLegacy/database/medLocalDbController.cpp:133]: same as previous comment

**Comment by mathildemerle** [comment]
@paulineMig i had a problem with my build, it's ok now, i can import et read, no problem on my side

**Review by mathildemerle — APPROVED** [review]

**Pull request merged**